### PR TITLE
anyrun: init at 0-unstable-2023-12-01

### LINKED ISSUE #255958: Package request: anyrun
**Project description**
 A wayland native, highly customizable runner. 

**Metadata**

* homepage URL: https://github.com/Kirottu/anyrun
* source URL: https://github.com/Kirottu/anyrun
* license: gpl3
* platforms: linux

### PR DESCRIPTION
**Pull request by eclairevoyant**
## Description of changes

Adds anyrun, a customizable app launcher for wayland

Homepage: https://github.com/Kirottu/anyrun/tree/master

Note, testing this requires creating `/etc/anyrun/style.css`

Fixes #255958.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/an/anyrun/package.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    gtk3
+    gtk-layer-shell
+    pango
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Security
+  ] ++ lib.optionals stdenv.isLinux [
+    wayland
+  ];
+
+  preFixup = ''
```
Comment (NotAShelf) [inline pkgs/by-name/an/anyrun/package.nix:54]: I'd rather if we built the plugins on demand, with something like `withPlugins ? false` that the user can override if they would like the plugins.

File: pkgs/by-name/an/anyrun/package.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    gappsWrapperArgs+=(
+     --prefix ANYRUN_PLUGINS : $out/lib
+    )
+  '';
+
+  meta = {
+    description = "A wayland-native, highly customizable runner";
+    homepage = "https://github.com/kirottu/anyrun";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ eclairevoyant ];
```
Comment (NotAShelf) [inline pkgs/by-name/an/anyrun/package.nix:64]: might as well add me here :p

File: pkgs/by-name/an/anyrun/package.nix (hunk 1 above)
Comment (eclairevoyant) [reply pkgs/by-name/an/anyrun/package.nix:54]: It seems that building without plugins breaks the program, unfortunately

File: pkgs/by-name/an/anyrun/package.nix [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+  passthru.updateScript = unstableGitUpdater { };
+
+  meta = {
+    description = "A wayland-native, highly customizable runner";
+    homepage = "https://github.com/kirottu/anyrun";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ eclairevoyant NotAShelf ];
+    mainProgram = "anyrun";
+    platforms = lib.platforms.linux ++ lib.platforms.darwin;
```
Comment (SuperSandro2000) [inline pkgs/by-name/an/anyrun/package.nix:73]:
```suggestion
```
that's the default for buildRustPackage and we should not repeat it

**Review by SuperSandro2000 — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/by-name/an/anyrun/package.nix [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    pango
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Security
+  ] ++ lib.optionals stdenv.isLinux [
+    wayland
+  ];
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+     --prefix ANYRUN_PLUGINS : $out/lib
```
Comment (Aleksanaa) [inline pkgs/by-name/an/anyrun/package.nix:57]:
```suggestion
     --prefix ANYRUN_PLUGINS : $out/lib
     --suffix XDG_DATA_DIRS: $out/share
```

```
=> ./result/bin/anyrun 
Failed to load custom CSS: <broken file>:1:0Failed to import: Error opening file /etc/anyrun/style.css: No such file or directory
Error reading applications plugin config: No such file or directory (os error 2)
Error reading directory /nix/store/c87saxq67wg8fwjn9cm9ahgapb09r87b-anyrun-0-unstable-2023-12-01/share: No such file or directory (os error 2)
Error reading directory /nix/store/8gdcfdqnj9rfbk2833vpbqbj6wc0vpvr-gsettings-desktop-schemas-45.0/share/gsettings-schemas/gsettings-desktop-schemas-45.0: No such file or directory (os error 2)
Error reading directory /nix/store/lk8kgif9ynk1l6aiqnbvzvys83n0h956-gtk+3-3.24.41/share/gsettings-schemas/gtk+3-3.24.41: No such file or directory (os error 2)
Error reading directory /home/aleksana/.nix-profile/share: No such file or directory (os error 2)
Error reading directory /nix/profile/share: No such file or directory (os error 2)
Error reading directory /home/aleksana/.local/state/nix/profile/share: No such file or directory (os error 2)
Error reading directory /nix/var/nix/profiles/default/share: No such file or directory (os error 2)
```

Comment (Aleksanaa) [reply pkgs/by-name/an/anyrun/package.nix:57]: not needed

**Pull request merged**